stl/inc/ranges

```diff
@@ -2490,6 +2490,412 @@ namespace ranges {
+
+    template <input_range _Vw, forward_range _Pat>
+        requires view<_Vw> && view<_Pat>
+            && indirectly_comparable<iterator_t<_Vw>, iterator_t<_Pat>, _RANGES equal_to>
+            && (forward_range<_Vw> || _Tiny_range<_Pat>)
+    class split_view;
+    // clang-format on
+
+    template <class _Vw, class _Pat>
+    class _Split_view_base : public view_interface<split_view<_Vw, _Pat>> {
+    protected:
+        /* [[no_unique_address]] */ iterator_t<_Vw> _Current{};
+    };
+    template <forward_range _Vw, class _Pat>
+    class _Split_view_base<_Vw, _Pat> : public view_interface<split_view<_Vw, _Pat>> {};
+
+    // clang-format off
+    template <input_range _Vw, forward_range _Pat>
+        requires view<_Vw> && view<_Pat>
+            && indirectly_comparable<iterator_t<_Vw>, iterator_t<_Pat>, _RANGES equal_to>
+            && (forward_range<_Vw> || _Tiny_range<_Pat>)
+    class split_view : public _Split_view_base<_Vw, _Pat> {
+        // clang-format on
+    private:
+        /* [[no_unique_address]] */ _Vw _Range{};
+        /* [[no_unique_address]] */ _Pat _Pattern{};
+
+        template <class>
+        struct _Outer_iter_base {};
+
+        // clang-format off
+        template <forward_iterator _Iter>
+        struct _Outer_iter_base<_Iter> {
+            // clang-format on
+            _Iter _Current{};
+        };
+
+        template <bool _Const>
+        class _Inner_iter;
+
+        template <bool _Const>
+        class _Outer_iter : private _Outer_iter_base<iterator_t<_Maybe_const<_Const, _Vw>>> {
+        private:
+            template <bool>
+            friend class _Inner_iter;
+            friend _Outer_iter<true>;
+
+            using _Mybase   = _Outer_iter_base<iterator_t<_Maybe_const<_Const, _Vw>>>;
+            using _ParentTy = _Maybe_const<_Const, split_view>;
+            using _BaseTy   = _Maybe_const<_Const, _Vw>;
+
+            _ParentTy* _Parent = nullptr;
+
+            _NODISCARD constexpr iterator_t<_BaseTy>& _Get_current() noexcept {
+                if constexpr (forward_range<_BaseTy>) {
+                    return this->_Current;
+                } else {
+                    return _Parent->_Current;
+                }
+            }
+
+            _NODISCARD constexpr const iterator_t<_BaseTy>& _Get_current() const noexcept {
+                if constexpr (forward_range<_BaseTy>) {
+                    return this->_Current;
+                } else {
+                    return _Parent->_Current;
+                }
+            }
+
+            _NODISCARD constexpr bool _At_end() const
+                noexcept(noexcept(_Implicitly_convert_to<bool>(_Get_current() == _RANGES end(_Parent->_Range)))) {
+                return _Get_current() == _RANGES end(_Parent->_Range);
+            }
+
+        public:
+            using iterator_concept  = conditional_t<forward_range<_BaseTy>, forward_iterator_tag, input_iterator_tag>;
+            using iterator_category = input_iterator_tag;

```

Comment:
Per WG21-P2259, this `iterator_category` is present only if _BaseTy models forward_range

Reply:
I hadn't planned to implement WG21-P2259 yet, but the wording changes for `split_view` are small and localized, so why not.

Reply:
Thanks, @cpplearner!